ReactWindows/ReactNative.Shared/UIManager/UIManagerModule.cs

```diff
@@ -125,17 +125,20 @@ public UIImplementation UIImplementation
         /// JavaScript can use the returned tag with to add or remove children 
         /// to this view through <see cref="manageChildren(int, int[], int[], int[], int[], int[])"/>.
         /// </remarks>
-        public int AddMeasuredRootView(ReactRootView rootView)
+        public async Task<int> AddMeasuredRootViewAsync(ReactRootView rootView)
         {
             // Called on main dispatcher thread
             DispatcherHelpers.AssertOnDispatcher();
 
             var tag = _nextRootTag;
             _nextRootTag += RootViewTagIncrement;
 
+            // Set tag early to accomodate raceing removals done by DetachRootViewAsync
```

Comment:
"racing" and "accommodate" or "Set tag early in case of concurrent DetachRootViewAsync"